smaps/src/main/java/com/google/smaps/Histogram.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.smaps;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Retrieves histogram information and formats it into a Json array for creating the chart
+ * and dashboard tools in histogram.js.
+ */
+@WebServlet(name = "Histogram", value = "/histogram")
+public class Histogram extends HttpServlet {
+  // Used to populate the lower bound text box and slider min.
+  static long lowerBound;
+  // Used to populate the upper bound text box and slider max.
+  static long upperBound;
+  // Flag indicating whether a user has entered info in textbox.
+  static boolean postFired = false;
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get numbers from form in interactive-histogram.html.
+    String lower = request.getParameter("lower-bound");
+    String upper = request.getParameter("upper-bound");
+    postFired = true;
+
+    // In case user entered a number with .0 as the decimal, cast to long to drop the decimal,
+    // otherwise just convert the String to a Long.
+    if (lower.contains(".")) {
```

Comment:
Are you doing this because Long.parseLong("4.5") throws an exception?

Reply:
I added some attributes to the HTML to only allow positive integers, but it still allows numbers with .0, which throws an exception, so that's why I added that.